qutip/solver/propagator.py

```diff
@@ -138,22 +145,25 @@ class Propagator:
         into a :class:`QobjEvo` with ::
             U = QobjEvo(Propagator(H))
     """
-    def __init__(self, H, c_ops=(), args=None, options=None,
+    def __init__(self, system, *, c_ops=(), args=None, options=None,
                  memoize=10, tol=1e-14):
-        Hevo = QobjEvo(H, args=args)
-        c_ops = [QobjEvo(op, args=args) for op in c_ops]
+        if isinstance(system, McSolver):
+            raise TypeError("McSolver cannot be used in propagator")
+        elif isinstance(system, Solver):
+            self.solver = system
+        else:
+            Hevo = QobjEvo(system, args=args)
+            c_ops = [QobjEvo(op, args=args) for op in c_ops]
+            if Hevo.issuper or c_ops:
+                self.solver = MeSolver(Hevo, c_ops=c_ops, options=options)
+            else:
+                self.solver = SeSolver(Hevo, options=options)
+
         self.times = [0]
         self.invs = [None]
-        if Hevo.issuper or c_ops:
-            self.props = [qeye(Hevo.dims)]
-            self.solver = MeSolver(Hevo, c_ops=c_ops, options=options)
-        else:
-            self.props = [qeye(Hevo.dims[0])]
-            self.solver = SeSolver(Hevo, options=options)
+        self.props = [qeye(self.solver.sys_dims)]
         self.cte = self.solver.rhs.isconstant
-        self.unitary = (not self.solver.rhs.issuper
-                        and isinstance(H, Qobj)
-                        and H.isherm)
+        self.unitary = isinstance(system, SeSolver)
```

Comment:
Is this correct? If the solver is created with a non-hermitian matrix, should the propagator _not_ be unitary?